lib/api_auth/request_drivers/faraday.rb

```diff
@@ -56,7 +56,9 @@ def content_md5
       end
 
       def request_uri
-        uri = URI::HTTP.new(nil, nil, nil, nil, nil, @request.path, nil, @request.params.to_query, nil)
+        query_string = @request.params.map{|k,v| "#{k}=#{v}"}.join('&')
```

Comment:
As far as I can tell, Faraday calls `to_query` before actually executing the request. https://github.com/lostisland/faraday/blob/81f16593a0138ec58bb6f25e1c2804e91589662f/lib/faraday/connection.rb#L407

So I think no mater what order you specify in the string when setting up the faraday connection, the query params will always be sorted when they hit the remote server.